lib/view/add_label_dialog.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+
+/// AddLabelDialog:
+///   - dialog for adding a label to a list (used by `ListSettingsDialog`)
+class AddLabelDialog extends StatefulWidget {
+  final Iterable<String> labels;
+
+  const AddLabelDialog(
+      {super.key, required this.labels});
+
+  @override
+  State<AddLabelDialog> createState() => _AddLabelDialogState();
+}
+
+class _AddLabelDialogState extends State<AddLabelDialog> {
+  String searchQuery = '';
+  late TextEditingController textEditingController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return SimpleDialog(
+      children: [
+        ListTile(
+            title: TextField(
+                autofocus: true,
+                controller: textEditingController,
+                decoration: InputDecoration(
+                    hintText: 'Enter a label',
+                    suffixIcon: IconButton(
+                        icon: Icon(Icons.add),
+                        onPressed: () => Navigator.pop(
+                            context, textEditingController.text))),
+                onChanged: (value) => setState(() => searchQuery = value),
+                onSubmitted: (newLabel) => Navigator.pop(context, newLabel))),
+        SizedBox(
+          height: 200,
```

Comment:
This doesn't need to be so tall. It could also have some top padding to give space between the first label `ListTile` and the text input.